app/services/payment_providers/stripe_service.rb

```diff
@@ -28,43 +27,16 @@ def create_or_update(**args)
       stripe_provider.success_redirect_url = args[:success_redirect_url] if args.key?(:success_redirect_url)
       stripe_provider.save!
 
-      if secret_key != stripe_provider.secret_key
-        unregister_webhook(stripe_provider, secret_key)
-
-        PaymentProviders::Stripe::RegisterWebhookJob.perform_later(stripe_provider)
-      end
-
       if payment_provider_code_changed?(stripe_provider, old_code, args)
         stripe_provider.customers.update_all(payment_provider_code: args[:code]) # rubocop:disable Rails/SkipsModelValidations
+        # Until this job is processed, the webhook endpoint will return 400 error
+        PaymentProviders::Stripe::RefreshWebhookJob.perform_later(stripe_provider)
```

Comment:
The provider code is passed as query param to the webhook endpoints to retrieve the correct connection.